WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift

```diff
@@ -127,6 +134,43 @@ private extension CardReaderSettingsConnectedViewController {
         tableView.reloadData()
     }
 
+    func configureUpdateView() {
+        guard viewModel != nil, let viewModelUpdateInProgress = viewModel?.readerUpdateInProgress else {
+            return
+        }
+        guard updateInProgress != viewModelUpdateInProgress else {
+            return
+        }
+        updateInProgress = viewModelUpdateInProgress
+
+        // If we are not updating a reader, dismiss any updateViewController
+        if !updateInProgress {
+            updateViewController?.dismiss(animated: true, completion: { [weak self] in
+                guard let self = self else {
+                    return
+                }
+                guard let readerUpdateCompletedSuccessfully = self.viewModel?.readerUpdateCompletedSuccessfully else {
+                    return
+                }
+                if readerUpdateCompletedSuccessfully {
+                    self.displayReaderUpdateSuccessNotice()
+                }
+
+                self.updateViewController = nil
```

Comment:
@ctarda - thoughts? I'd like to completely nuke the view controller when I'm done with it. Although this works, is it a good idea?

Reply:
The [documentation about dismiss](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621505-dismiss) says:

> If you want to retain a reference to the view controller's presented view controller, get the value in the presentedViewController property before calling this method.

Which I interpret as "the reference to the view controller will become nil eventually". I don't think it's strictly necessary to nullify this reference, but it might not hurt do to so?